database/database.go

```diff
@@ -237,3 +251,76 @@ func connectionString(creds DBCredentials) string {
 		mongoWriteConcernTimeout,
 	)
 }
+
+// ensureDBSchema checks that we have all collections and indexes we need and
+// creates them if needed.
+// See https://docs.mongodb.com/manual/indexes/
+// See https://docs.mongodb.com/manual/core/index-unique/
+func ensureDBSchema(ctx context.Context, db *mongo.Database) error {
+	// schema defines a mapping between a collection name and the indexes that
+	// must exist for that collection.
+	schema := map[string][]mongo.IndexModel{
+		dbUsersCollection: {
+			{
+				Keys:    bson.D{{"sub", 1}},
+				Options: options.Index().SetName("usersSubUnique").SetUnique(true),
+			},
+		},
+		dbSkylinksCollection: {
+			{
+				Keys:    bson.D{{"skylink", 1}},
+				Options: options.Index().SetName("skylinksSkylinkUnique").SetUnique(true),
+			},
+		},
+		dbUploadsCollection: {
+			{
+				Keys:    bson.D{{"user_id", 1}},
```

Comment:
(nit) are all fields in our documents snakecase? I would've expected `userId` here actually. I would actually change these to `user` or `userId` because we'll be accessing them in objects and using them in queries. We'll end up with stuff like 
```
const upload = await uploadsDB.findBy(...);
console.log(upload.user_id)
```

in any case we should be consistent, if we're consistent I'm no tbothered by it

Reply:
The fields are snake case in Mongo but not in Go. Here's the `Upload` type:
```go
type Upload struct {
	ID        primitive.ObjectID  `bson:"_id,omitempty" json:"_id"`
	UserID    primitive.ObjectID  `bson:"user_id,omitempty" json:"user_id"`
	SkylinkID primitive.ObjectID  `bson:"skylink_id,omitempty" json:"skylink_id"`
	Timestamp primitive.Timestamp `bson:"timestamp" json:"timestamp"`
}
```
So, this is how the usage looks like:
```go
func (db *DB) UploadsByUser(ctx context.Context, user User) ([]Upload, error) {
	if user.ID.IsZero() {
		return nil, errors.New("invalid user")
	}
	filter := bson.D{{"user_id", user.ID}}
	c, err := db.staticUploads.Find(ctx, filter)
	if err != nil {
		return nil, err
	}
	uploads := make([]Upload, 0)
	err = c.All(ctx, &uploads)
	if err != nil {
		return nil, err
	}
	return uploads, nil
}
```
And then you can use the resulting `[]Upload` in a way idiomatic to Go.